crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -133,18 +133,31 @@ where
                         format!("101,{child_id},{child_id},thin-edge.io-child"),
                     ));
                 }
-
-                vec.push(Message::new(
-                    &self.mapper_config.out_topic,
-                    c8y_json_child_payload,
-                ));
+                if c8y_json_child_payload.len() < self.size_threshold.0 {
+                    vec.push(Message::new(
+                        &self.mapper_config.out_topic,
+                        c8y_json_child_payload,
+                    ));
+                } else {
+                    return Err(ConversionError::C8yMqttPayloadThresholdExceed(
+                        c8y_json_child_payload[0..50].into(),
+                        c8y_json_child_payload.len(),
+                    ));
+                }
```

Comment:
This piece of code is duplicated for the root vs child device cases. But the logic is the same in both case.

Consider to have something along the lines:

```
let c8y_json_payload = match maybe_child_id {
     Some(child_id) => {
        let c8y_json_child_payload =
                    json::from_thin_edge_json_with_child(input.payload_str()?, child_id.as_str())?;
        if !self.children.contains(child_id.as_str()) {
             // ...
        }
        c8y_json_child_payload
     },
     None => json::from_thin_edge_json(input.payload_str()?)?
};

if c8y_json_payload.len() < self.size_threshold.0 {
   // ...
} else {
   // ..
}
```

Reply:
refactored